common/components/controllers/SignUpController.jsx

```diff
@@ -13,19 +16,51 @@ type State = {|
   email: string,
   password1: string,
   password2: string,
+  validations: $ReadOnlyArray<Validation>,
+  isValid: boolean
 |}
 
 class SignUpController extends React.Component<Props, State> {
   constructor(): void {
     super();
+    
     this.state = {
       firstName: '',
       lastName: '',
       email: '',
       password1: '',
       password2: '',
+      isValid: false,
+      validations: [
+        {
+          checkFunc: (state: State) => !_.isEmpty(state.firstName),
+          errorMessage: "Please enter First Name"
+        },
+        {
+          checkFunc: (state: State) => !_.isEmpty(state.lastName),
+          errorMessage: "Please enter Last Name"
+        },
+        {
+          checkFunc: (state: State) => !_.isEmpty(state.email),
+          errorMessage: "Please enter email address"
+        },
+        {
+          checkFunc: (state: State) => state.password1.length >= 8,
```

Comment:
this magic number of `8` should probably be hoisted to a class constant if not perhaps even higher to an app constant

Reply:
Done